engine/collection/ingest/engine.go

```diff
@@ -201,14 +209,15 @@ func (e *Engine) onTransaction(originID flow.Identifier, tx *flow.TransactionBod
 
 	// if our cluster is responsible for the transaction, add it to the mempool
 	if localCluster.Fingerprint() == txCluster.Fingerprint() {
-		_ = e.pool.Add(tx)
-		e.colMetrics.TransactionIngested(tx.ID())
+		_ = pool.Add(tx)
+		e.colMetrics.TransactionIngested(txID)
 		log.Debug().Msg("added transaction to pool")
 	}
 
 	// if the message was submitted internally (ie. via the Access API)
-	// propagate it to all members of the responsible cluster
-	if originID == localID {
+	// propagate it to members of the responsible cluster (either our cluster
```

Comment:
@jordanschalm do you think the access node should also do this - forward the transaction to the cluster based on an epoch. Currently, it does not account for epoch.

Reply:
I think it does already:
https://github.com/onflow/flow-go/blob/1e0a30e7a9118aad96a706639e37ec955f562542/engine/access/rpc/backend/backend_transactions.go#L107-L108

It might have been updated along with the epoch-aware protocol state changes.

Reply:
oh cool. thanks!